packages/lab/src/number-input/NumberInput.tsx

```diff
@@ -246,139 +271,150 @@ export const NumberInput = forwardRef<HTMLDivElement, NumberInputProps>(
       ? ["required", "asterisk"].includes(formFieldRequired)
       : inputRequired;
 
-    const isAdjustingRef = useRef<boolean>(false);
-    const [isEditing, setIsEditing] = useState(false);
     const [isFocused, setIsFocused] = useState(false);
-
-    const [recordCaret, restoreCaret, resetCaret] = useCaret({
-      inputRef,
-    });
+    const [isEditing, setIsEditing] = useState(false);
+    const { DecreaseIcon, IncreaseIcon } = useIcon();
 
     const [value, setValue] = useControlled({
-      controlled: valueProp,
-      default: defaultValue,
+      controlled: valueProp !== undefined ? String(valueProp) : undefined,
+      default: String(defaultValueProp ?? ""),
       name: "NumberInput",
       state: "value",
     });
+    const lastNumberChange = useRef<{
+      value: number | null;
```

Comment:
```suggestion
      parsedValue: number | null;
```
Nit (not mandatory): Should `value` be renamed to something more descriptive?